plugins/ksr/endpoints_reflector.go

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) 2017 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ksr
+
+import (
+	"reflect"
+	"sync"
+
+	coreV1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/client-go/tools/cache"
+
+	proto "github.com/contiv/vpp/plugins/ksr/model/endpoints"
+)
+
+// EndpointsReflector subscribes to K8s cluster to watch for changes
+// in the configuration of k8s service endpoints.
+// Protobuf-modelled changes are published into the selected key-value store.
+type EndpointsReflector struct {
+	ReflectorDeps
+
+	stopCh                 <-chan struct{}
+	wg                     *sync.WaitGroup
+	k8sEndpointsStore      cache.Store
+	k8sEndpointsController cache.Controller
+	stats                  ReflectorStats
+}
+
+// Init subscribes to K8s cluster to watch for changes in the configuration
+// of k8s services. The subscription does not become active until Start()
+// is called.
+func (epr *EndpointsReflector) Init(stopCh2 <-chan struct{}, wg *sync.WaitGroup) error {
+	epr.Log.Info("EndpointsReflector Init()")
+	epr.stopCh = stopCh2
+	epr.wg = wg
+
+	restClient := epr.K8sClientset.CoreV1().RESTClient()
+	listWatch := epr.K8sListWatch.NewListWatchFromClient(restClient, "endpoints", "", fields.Everything())
+	epr.k8sEndpointsStore, epr.k8sEndpointsController = epr.K8sListWatch.NewInformer(
+		listWatch,
+		&coreV1.Endpoints{},
+		0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				eps, ok := obj.(*coreV1.Endpoints)
+				if !ok {
+					epr.Log.Warn("Failed to cast newly created endpoints object")
+					epr.stats.NumArgErrors++
+				} else {
+					epr.addEndpoints(eps)
+				}
+			},
+			DeleteFunc: func(obj interface{}) {
+				eps, ok := obj.(*coreV1.Endpoints)
+				if !ok {
+					epr.Log.Warn("Failed to cast removed endpoints object")
+					epr.stats.NumArgErrors++
+				} else {
+					epr.deleteEndpoints(eps)
+				}
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				epsOld, ok1 := oldObj.(*coreV1.Endpoints)
+				epsNew, ok2 := newObj.(*coreV1.Endpoints)
+				if !ok1 || !ok2 {
+					epr.Log.Warn("Failed to cast changed endpoints object")
+					epr.stats.NumArgErrors++
+				} else {
+					epr.updateEndpoints(epsNew, epsOld)
+				}
+			},
+		},
+	)
+	return nil
+}
+
+// GetStats returns the Endpoints Reflector usage statistics
+func (sr *EndpointsReflector) GetStats() *ReflectorStats {
+	return &sr.stats
```

Comment:
travis:  receiver name sr should be consistent with previous receiver name epr for EndpointsReflector

Otherwise, the code looks good to me.